pkg/scheduler/internal/parallelize/parallelism.go

```diff
@@ -23,13 +23,15 @@ import (
 	"k8s.io/client-go/util/workqueue"
 )
 
-const parallelism = 16
+// Parallelism is exported for tests where non-determinism from parallelism is
+// not desired. Do NOT modify outside tests.
+var Parallelism = 16
```

Comment:
Could we avoid this by:

1. Having a test for 100% of nodes.
2. Having another test for different percentages, validate the size of the list and makes sure all the elements belong to the known list of all preemptible Nodes.

Reply:
I don't think it's worth it IMHO. The size of the list isn't deterministic (in spite of cancelling the context, some dry runs could already be in-flight) and the elements of the list aren't deterministic. Due to the small number of nodes in tests (5) and the large parallelism (16), we would need to add all nodes to the acceptable list of candidates, which defeats the purpose of the test entirely.

Reply:
If a dry run is already in-flight, it's result would be discarded because the limited was exceeded, or not?

Reply:
It wouldn't be discarded. Once a dry run begins, if it can fit the preemptor, it will be added to one of the lists. And that is fine, `numCandidates` is only an approximation; it doesn't really matter if we dry run 100 nodes or 102 nodes.

Reply:
I see. FYI @hprateek43 for #94636

Reply:
I'm ok with the change, it's just that in general we should think of all our options before exposing something just for tests, specially if it may change the behavior. The test can be designed to allow non-deterministic results.

Reply:
Yep, I agree that we can test non-deterministic results with some significant test refactor, but the benefit vs cost is just too high to justify it IMO. Besides, I think there will be a cleaner API to set parallelism once https://github.com/kubernetes/kubernetes/pull/94636 is merged.

Reply:
can you add a todo to make this private again once #94636 merges (or as part of that PR?)